open_bucket_hashtable.h

```diff
@@ -186,11 +186,16 @@ template <typename GID, typename LID, int maxBucketOverflow = 4, GID EMPTYBUCKET
       iterator(OpenBucketHashtable<GID, LID>& hashtable, size_t index) : hashtable(&hashtable), index(index) {}
 
       iterator& operator++() {
-         do {
+         index++;
+         while(index < hashtable->buckets.size()){
+            if (hashtable->buckets[index].first != EMPTYBUCKET){
+               break;
+            }
             index++;
-         } while (hashtable->buckets[index].first == EMPTYBUCKET && index < hashtable->buckets.size());
+         }
          return *this;
       }
+      
```

Comment:
Isn't the new code here doing exactly the same as the old one?
Index gets incremented at least once, and then exactly the same conditions break out of the loop.

Reply:
The check against buckets.size() is now separate and before the check against EMPTYBUCKET, preventing out-of-bounds.

Reply:
Yep as Markus said